### PR TITLE
refactor: rename Batch to BatchLockup

### LINKED ISSUE #320: Rename Batch to BatchLockup
### Discussed in https://github.com/sablier-labs/v2-periphery/discussions/318

<div type='discussions-op-text'>

<sup>Originally posted by **andreivladbrg** April 10, 2024</sup>
With the new OE contract, we will have multiple batch contracts, but in different repos.

Thus, to differentiate between them, shouldn't we rename the batch contract to: `SablierV2BatchLockup`?

@sablier-labs/solidity 

Also, tagging @razgraf to ask if this would improve the dev work at the app level?</div>

### PR DESCRIPTION
**Pull request by smol-ninja**
closes #320

### CODE REVIEW

File: test/fork/assets/USDC.t.sol [hunk 1]
```diff
@@ -3,25 +3,26 @@ pragma solidity >=0.8.22 <0.9.0;
 
 import { IERC20 } from "@openzeppelin/contracts/token/ERC20/IERC20.sol";
 
-import { CreateWithTimestamps_LockupDynamic_Batch_Fork_Test } from "../batch/createWithTimestampsLD.t.sol";
-import { CreateWithTimestamps_LockupLinear_Batch_Fork_Test } from "../batch/createWithTimestampsLL.t.sol";
-import { CreateWithTimestamps_LockupTranched_Batch_Fork_Test } from "../batch/createWithTimestampsLT.t.sol";
+import { CreateWithTimestamps_LockupDynamic_Batch_Lockup_Fork_Test } from "../batch-lockup/createWithTimestampsLD.t.sol";
+import { CreateWithTimestamps_LockupLinear_Batch_Lockup_Fork_Test } from "../batch-lockup/createWithTimestampsLL.t.sol";
+import { CreateWithTimestamps_LockupTranched_Batch_Lockup_Fork_Test } from
```
Comment (PaulRBerg) [inline test/fork/assets/USDC.t.sol:8]: `BatchLockup` is the name of a single contract so let's remove the underscore between these words.

File: test/utils/BatchLockupBuilder.sol [hunk 2]
```diff
@@ -0,0 +1,232 @@
+// SPDX-License-Identifier: GPL-3.0-or-later
+pragma solidity >=0.8.22;
+
+import { LockupDynamic, LockupLinear, LockupTranched } from "@sablier/v2-core/src/types/DataTypes.sol";
+
+import { BatchLockup } from "../../src/types/DataTypes.sol";
+
+library BatchLockupBuilder {
+    /// @notice Generates an array containing `batchSize` copies of `batchSingle`.
+    function fillBatchLockup(
```
Comment (PaulRBerg) [inline test/utils/BatchLockupBuilder.sol:10]: Adding  `Lockup` is redundant in these functions. That this is about `BatchLockup` is implied by the name of the library (`BatchLockupBuilder`).

File: test/utils/Defaults.sol [hunk 3]
```diff
@@ -376,63 +376,87 @@ contract Defaults is Merkle {
...
     /*//////////////////////////////////////////////////////////////////////////
-                                        BATCH
+                                    BATCH-LOCKUP
     //////////////////////////////////////////////////////////////////////////*/
 
-    /// @dev Returns a default-size batch of `Batch.CreateWithDurationsLD` parameters.
-    function batchCreateWithDurationsLD() public view returns (Batch.CreateWithDurationsLD[] memory batch) {
-        batch = BatchBuilder.fillBatch(createWithDurationsLD(), BATCH_SIZE);
+    /// @dev Returns a default-size batch of `BatchLockup.CreateWithDurationsLD` parameters.
+    function batchLockupCreateWithDurationsLD()
```
Comment (PaulRBerg) [inline test/utils/Defaults.sol:383]: That this is about `Lockup` is implied by `LD`. We can just say `batchCreateWithDurationsLD`, and ditto for the other functions in here.

File: test/utils/DeployOptimized.sol [hunk 4]
```diff
@@ -21,9 +21,9 @@ abstract contract DeployOptimized is StdCheats {
 
     /// @notice Deploys all V2 Periphery contracts from a optimized source in the following order:
```
Comment (PaulRBerg) [inline test/utils/DeployOptimized.sol:22]: "an optimized source"

**Review by PaulRBerg — CHANGES_REQUESTED** [review]
changes requested below

**Review by PaulRBerg — APPROVED** [review]
I have addressed all of my feedback above in the latest commit. LGTM now.

**Comment by smol-ninja** [comment]
Similar to https://github.com/sablier-labs/v2-periphery/pull/321, I will be re-doing it with `prb-feedback` changes as there are a so many git conflicts.

**Comment by smol-ninja** [comment]
@PaulRBerg can you please have a quick look to this as well?

File: test/utils/BatchLockupBuilder.sol [hunk 5]
```diff
@@ -3,35 +3,35 @@ pragma solidity >=0.8.22;
...
     /// @notice Generates an array containing `batchSize` copies of `batchSingle`.
     function fillBatch(
-        Batch.CreateWithDurationsLD memory batchSingle,
+        BatchLockup.CreateWithDurationsLD memory batchSingle,
         uint256 batchSize
     )
         internal
         pure
-        returns (Batch.CreateWithDurationsLD[] memory batch)
+        returns (BatchLockup.CreateWithDurationsLD[] memory batchLockup)
```
[the remark this answers is not in the document]
Comment (smol-ninja) [reply test/utils/BatchLockupBuilder.sol:16]: Cool. Making changed now.

**Pull request merged**